src/sphinx_autodoc_typehints/__init__.py

```diff
@@ -89,7 +89,21 @@ def get_annotation_args(annotation: Any, module: str, class_name: str) -> tuple[
     return getattr(annotation, "__args__", ())
 
 
-def format_annotation(annotation: Any, fully_qualified: bool = False, simplify_optional_unions: bool = True) -> str:
+def format_annotation(
+    annotation: Any,
+    fully_qualified: bool = False,
+    simplify_optional_unions: bool = True,
+    typehints_formatter: Callable[..., str] | None = None,
+) -> str:
+    if typehints_formatter is not None:
+        formatted = typehints_formatter(
+            annotation,
+            fully_qualified=fully_qualified,
+            simplify_optional_unions=simplify_optional_unions,
```

Comment:
On second thought let's pass in the `app.config` argument here as is. That way we'll not have to keep adding arguments in the future.